### PR TITLE
gh-2625: Improve assertions readability with assertJ

### PR DESCRIPTION
**Pull request by AlfonsoEsposito**

### CODE REVIEW

File: core/common-util/src/test/java/uk/gov/gchq/gaffer/commonutil/iterable/CachingIterableTest.java [hunk 1]
```diff
@@ -51,8 +50,9 @@ public void shouldCacheSmallIterable(@Mock final Iterable<Integer> mockIterable)
 
         final Iterable<Integer> cachingIterable = new CachingIterable<>(mockIterable, 5);
 
-        assertThat(SMALL_LIST).isEqualTo(Lists.newArrayList(cachingIterable));
-        assertThat(SMALL_LIST).isEqualTo(Lists.newArrayList(cachingIterable));
+        assertThat(cachingIterable)
+                .containsExactlyElementsOf(SMALL_LIST)
+                .containsExactlyElementsOf(SMALL_LIST);
```
Comment (GCHQDeveloper314) [inline core/common-util/src/test/java/uk/gov/gchq/gaffer/commonutil/iterable/CachingIterableTest.java:55]: I'm not sure why this was duplicated previously, but I can't see a reason for it to remain. Same for the `LARGE_LIST ` assertion below.

**Review by GCHQDeveloper314 — CHANGES_REQUESTED** [review]
This doesn't address everything required by the issue. The original issue comment only gave indicative examples of the work required and not which files need to be changed.

[I've added a new comment to the issue](https://github.com/gchq/Gaffer/issues/2625#issuecomment-1687723803) with further information and some hints on where changes need to be made.

File: core/common-util/src/test/java/uk/gov/gchq/gaffer/commonutil/iterable/CachingIterableTest.java (hunk 1 above)
Comment (t92549) [reply core/common-util/src/test/java/uk/gov/gchq/gaffer/commonutil/iterable/CachingIterableTest.java:55]: I think it is to test that the iterable doesn't get exhausted after being evaluated

**Comment by t92549** [comment]
@GCHQDeveloper314 Although this will not close the issue, it does make progress towards and so I think can be merged

File: core/common-util/src/test/java/uk/gov/gchq/gaffer/commonutil/iterable/CachingIterableTest.java (hunk 1 above)
Comment (GCHQDeveloper314) [reply core/common-util/src/test/java/uk/gov/gchq/gaffer/commonutil/iterable/CachingIterableTest.java:55]: Ok, in that case how sure are we sure that these assertions can test for that?

**Review by GCHQDeveloper314 — APPROVED** [review]

**Review by t92549 — APPROVED** [review]

**Pull request merged**